numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4166,6 +4166,55 @@ small_correlate(const char * d_, npy_intp dstride,
     }
 }
 
+/*
+*/
+
+/* A clone function for the datetime dtype metadata */
+static NpyAuxData *
```

Comment:
Any idea why this one is static, and the other is NPY_NO_EXPORT?

Reply:
Something about an old compiler version, looking at the referenced issue #5163. We can try to make the changed function static and see if we get crash reports. I thought marking it with `NPY_NO_EXPORT` would not export, but see issue #12438, it seems that macro turns into the empty sting

Reply:
Oh, I hadn't realized that comment was describing the function itself - I though it was a remark about allocating the metadata statically. Fine to leave as is, and revisit some other time.